crates/holochain_sqlite/src/sql/cell/action_hash_by_prev.sql

```diff
@@ -0,0 +1,19 @@
+SELECT
+  hash,
+  blob
+FROM
+  Action
+WHERE
+  prev_hash = :prev_hash
+  AND CASE
+    WHEN -- When this is an Update action for an Agent key...
+    entry_type = :entry_type_agent THEN -- ...the prev action's author will be different from the following action's author,
```

Comment:
It seems this is not hooked up yet as I don't see `:entry_type_hash` as a parameter yet.

Anyhow, what I meant is actually something else. Not the updates of agent keys, but catering for any action after an agent key has been updated. Say I have `agent_key_1` and create some entries, then update my key to `agent_key_2` and create another few entries. To pass sys validation, `detect_fork` needs to check for the new agent key after the key rotation.

But thinking about all this, it seems to me like it's enough to check in the SQL for an action that has the same previous hash than the action under validation. Source chains don't have multiple authors any more (except for key rotation, but that must be handled in Rust and not SQL) and with capabilities you're writing to the chain under the author's key.

Reply:
Wow, it's confusing that I wrote a detailed test that tests what I wanted, which still passes even when this query wasn't fully hooked up. I implemented it so wrong that it's right.

Yes, my mistake in writing this query is that I meant to check that the previous action is an Update, not the fork candidate. So I'm going to rewrite this now.